derp/derp_test.go

```diff
@@ -1598,3 +1599,19 @@ func TestServerRepliesToPing(t *testing.T) {
 		}
 	}
 }
+
+func TestGetPerClientSendQueueDepth(t *testing.T) {
+	envKey := "DERP_PER_CLIENT_SEND_QUEUE_DEPTH"
+	val, ok := os.LookupEnv(envKey)
+	if !ok {
+		defer os.Unsetenv(envKey)
+	} else {
+		defer os.Setenv(envKey, val)
+	}
+
+	os.Unsetenv(envKey)
+	assert.Equal(t, defaultPerClientSendQueueDepth, getPerClientSendQueueDepth())
+
+	t.Setenv(envKey, "64")
+	assert.Equal(t, 64, getPerClientSendQueueDepth())
```

Comment:
maybe https://pkg.go.dev/testing#T.Setenv ?

Reply:
I could not find a way to unsetenv with `T.` in my searches.

Reply:
It does it automatically when a test ends. So you could probably use subtests and t.Setenv within the subtests.

Reply:
But to clear the env, would `t.Setenv(envKey, "")` be equivilent to `os.UnsetEnv`?

Reply:
I'm wanting to specifically test the unset case and ensure it's clear for the test.

Reply:
Despite the name `envknob.LookupInt`, it doesn't distinguish between empty and unset like os.LookupEnv does...

```go
func LookupInt(envVar string) (v int, ok bool) {
	assertNotInInit()
	val := os.Getenv(envVar)
	if val == "" {
		return 0, false
	}
	v, err := strconv.Atoi(val)
	if err == nil {
		noteEnv(envVar, val)
		return v, true
	}
...
```

So setting it to the empty string is fine.

But either way. I was just pointing out T.Setenv because it's new-ish.